src/responder/component.ts

```diff
@@ -54,8 +54,13 @@ export class ResponderAppointmentReducer implements StateReducer<ResponderAppoin
             if (!tx.to) continue;
 
             // look for matching transactions
-            const txIdentifier = new PisaTransactionIdentifier(tx.chainId, tx.data, tx.to, tx.value, tx.gasLimit);
+            // we should use chain id, but for some reason chain id is not present in transactions from ethersjs
+            // therefore we fallback to network id when chain id is not present
+            const txChainId = tx.chainId || (tx as any).networkId;
```

Comment:
It looks like ethers.js declares `chainId` in the type, but it does not sanitizes if it receives `networkId` from the nodes (not sure why the inconsistency).
Not a big fan of having this here, I would rather sanitize it in the BlockFactory of the BlockProcessor, setting the chainId as we expect if networkId is set instead.

Reply:
Hmm, yeah, the chainid thing is really annoying tbh.

The real problem I have here is that network id and chain id are actually different - I've probably made a bad decision to use network id as a fallback. Perhaps a better way would be to get the chain id from the provider, since we know if we got a transaction from there it must have the same chain id.

Reply:
I've seen ethers.js' code during the review, and there was a place were networkId was assigned from chainId, or vice versa - so yeah, it might be better to avoid relying on ethers.js for this!

Reply:
I've kept the code and moved it down into the block factory. Although it is dodgy to interchange the two, as long as we never point at Ethereum Classic or any other forks this shouldnt be a problem. Those sound like famous last words lol.